### PR TITLE
adding DEC operators

### PR DESCRIPTION
**Pull request by odedstein**
This pull request adds DEC operators in the style of https://www.cs.cmu.edu/~kmcrane/Projects/DDG/

@sgsellan I'd be glad if you could double check to make sure I did these correctly :)

### CODE REVIEW

**Review by sgsellan — COMMENTED** [review]
This PR breaks our (unofficial) "one function per .py file" rule. Do we want to break this?

**Comment by odedstein** [comment]
I don't care whether we have separate files for this or not, I have no strong feelings.
You can decide.

File: src/gpytoolbox/dec_h1.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    --------
+    ```python
+    # Mesh in V,F
+    h1inv = gpy.dec_h1inv(V,F)
+    ```
+    
+    """
+
+    l_sq = halfedge_lengths_squared(V,F)
+    return dec_h1_intrinsic(l_sq,F,E_to_he=E_to_he)
```
Comment (sgsellan) [inline src/gpytoolbox/dec_h1.py:83]: Aha! Shouldn't this be `dec_h1inv_intrinsic`?

**Comment by sgsellan** [comment]
See typo above. We did not catch that typo due to a bigger problem: unless I am missing something, currently, the `inv` functions (intrinsic and not) have no unit test.

**Comment by odedstein** [comment]
Good point. Will fix this and add unit tests.

**Comment by sgsellan** [comment]
This looks good to me now! Happy to merge

**Pull request merged**